### PR TITLE
[WebHost] Unify style and behavior of popover and mobile menus

### PR DESCRIPTION
**Pull request by LegendaryLinux**
## What is this fixing or adding?
Based on feedback from the designer, the styles of the popover and mobile menus have been unified.
Popover and mobile menus are now toggled by clicking on their respective buttons, and may be closed by clicking anywhere.

## How was this tested?
Tested on Chrome and Firefox on both desktop and Android.

## If this makes graphical changes, please attach screenshots.
Desktop view in 1080p:
image

Desktop and mobile view at < 1580px width:
image

Desktop view at < 960px width:
image

Mobile view in portrait:
Screenshot_20230325-144734

### CODE REVIEW

File: WebHostLib/templates/header/baseHeader.html [hunk 1]
```diff
@@ -11,18 +11,16 @@
...
-                    <div><a href="/uploads">upload game</a></div>
-                    <div><a href="/user-content">user content</a></div>
-                    <div><a href="/tutorial">setup guides</a></div>
-                </span>
+            <div id="base-header-popover-text">
+                <img id="base-header-popover-icon" src="/static/static/button-images/popover.png" alt="Popover Menu" />
+                start playing
+            </div>
+            <div id="base-header-popover-menu">
+                <a class="first-link" href="/games">supported games</a>
```
Comment (black-sliver) [inline WebHostLib/templates/header/baseHeader.html:19]: why not `:first-child`?

File: WebHostLib/templates/header/baseHeader.html [hunk 2]
```diff
@@ -34,12 +32,12 @@
         </div>
     </header>
     <div id="base-header-mobile-menu">
-        <a href="/games">supported games</a>
+        <a class="first-link" href="/games">supported games</a>
```
Comment (black-sliver) [inline WebHostLib/templates/header/baseHeader.html:35]: as above

Comment (LegendaryLinux) [reply WebHostLib/templates/header/baseHeader.html:35]: That's a better idea, in case they get changed later.

**Comment by black-sliver** [comment]
Played around a little and came across the following things:
* Compatibility (I don't know why it happens or if they are bogus):
    * Selecting Galaxy S10 in Firefox' responsive design view has a broken zoom level for whatever reason
    * Selecting iPadOS in Firefox' responsive design view has the footer not at the bottom
* Style (my personal opinion, I don't claim any authority):
    * Not having any border or shadow on the right side of the pop over looks odd to me, especially when there is a cloud crossing below it
    * The blue shadow looks odd when it ends on top of an island (I personally would've picked `RGBA(0,0,0,0.5)` or something)
    * The mobile portrait menu items have a very big text. While it's fine space-wise, they appear out of proportion to my eyes compared to the hamburger and logo text height.
* Consistency / usability:
    * On the big layout, the "Start Playing"-Island and the "Start Playing"-Menu item now mean/provide two different things. I think one should be renamed so they appear with different names to the user, but I don't have a good suggestion.
    * "Upload Host File" sounds odd. We call it "host a pre-generated game" on the old page; maybe just "Host a game" would be clear enough if "Generate Game" is right next to it.

**Comment by LegendaryLinux** [comment]
Oh yeah, the responsive design view is going to give you some false positives. I am using `@media only screen and (max-device-width: 768px) as my selector for the mobile-only styles. Unless you click "Enable Touch Simulation" the browser will continue to report it has extra devices, and the `only screen` rule is ignored. So you get some odd things.

There is also an inconsistency in that view. If you shrink the responsive view too thin in desktop, it gets weird with the header:
image

But if you view it outside the responsive view, the header works just fine:
image

In any case, you are not the first one to note the text size difference on the mobile popover buttons. I'll try to poke the designer about it. She was the one who originally wanted all the extra padding.

And good call on the Start Playing button. I'll get that sorted.
As for the "Generate Game / Host Game" wording, this has been debated endlessly with no real consensus on best phrasing.

**Comment by LegendaryLinux** [comment]
Oh, and regarding the odd looking popover menu style on the right, I have to keep telling everyone. It's a placeholder until the designer can make cloud menu assets. She's booked through the end of April, currently.

**Comment by LegendaryLinux** [comment]
I changed the popover trigger text to "Get Started." I also tinkered with some CSS @black-sliver, and I came up with something that looks noticeably better on the popover styles when it covers a cloud:
image

And finally, I adjusted the font-size and padding on the mobile menu. Hopefully people stop complaining about it now:
image

**Comment by black-sliver** [comment]
I'll take a closer look later (in about 3-4 hours)

**Comment by kindasneaki** [comment]
You should probably change the mobile link to `host game` As well if that’s what you are sticking with. I still like `upload game` more but that’s just my opinion.

**Review by black-sliver — APPROVED** [review]
> Oh yeah, the responsive design view is going to give you some false positives.

Yeah, in the worst case it hopefully just behaves as before and we get a bug report.

> Oh, and regarding the odd looking popover menu style on the right, I have to keep telling everyone. It's a placeholder until the designer can make cloud menu assets. She's booked through the end of April, currently.

Having an intermediate solution is fine by me, just wanted to bring it up and the faint "border" already helps a bit.


Looks better to me now. Can only wait for more feedback at this point (by hosting it on the RC or the regular site).

**Pull request merged**